sdk/core/azure-core/inc/azure/core/cryptography/hash.hpp

```diff
@@ -60,15 +62,10 @@ namespace Azure { namespace Core { namespace Cryptography {
      */
     void Append(const uint8_t* data, std::size_t length)
     {
-      if (!data && length != 0)
-      {
-        throw std::invalid_argument(
-            "Length cannot be " + std::to_string(length) + " if the data pointer is null.");
-      }
-      if (m_isDone)
-      {
-        throw std::runtime_error("Cannot call Append after calling Final().");
-      }
+      AZURE_ASSERT_MSG(
+          (data && length >= 0) || (!data && length == 0),
+          "Length cannot be negative, and length must be 0 if the data pointer is null.");
```

Comment:
`length >= 0` is a tautology; `std::size_t` cannot represent negative values. Also, you've strengthened the condition here to forbid `data != nullptr && length == 0` which seems like a perfectly valid empty range.
```suggestion
      AZURE_ASSERT_MSG(
          data != nullptr || length == 0,
          "Length must be 0 if the data pointer is null.");
```

Reply:
> `length >= 0` is a tautology; `std::size_t` cannot represent negative values.

Agreed, already fixed. The change you were looking at is outdated.

> Also, you've strengthened the condition here to forbid `data != nullptr && length == 0` which seems like a perfectly valid empty range.

That was intentional, but let me revisit that to make sure it is what we want. Good call out :)

Reply:
> Also, you've strengthened the condition here to forbid `data != nullptr && length == 0` which seems like a perfectly valid empty range.

With the original assert (`(data && length >= 0) || (!data && length == 0)`), I don't think I had. Isn't that identical? That said, yours is simpler.